### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by patrys**
* Enabled type checking in VSCode
* Various small fixes in type hints
* Fixed a flaky address generator

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/core/tests/test_core.py [hunk 1]
```diff
@@ -248,14 +248,13 @@ def test_validate_image_file():
...
-    img = SimpleUploadedFile("product.jpg", img_data.getvalue(), "image/jpeg")
     field = "image"
 
     # when
-    result = validate_image_file(img, field, ProductErrorCode)
+    img = SimpleUploadedFile("product.jpg", img_data.getvalue(), "image/jpeg")
 
     # then
-    assert not result
+    validate_image_file(img, field, ProductErrorCode)
```
Comment (IKarbowiak) [inline saleor/graphql/core/tests/test_core.py:257]: `validate_image_file` is a method that we are testing not `SimpleUploadedFile`, so IMO it should go to `# when` part.

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — COMMENTED** [review]
Maybe we can also fix this: https://github.com/saleor/saleor/discussions/11044

File: saleor/graphql/core/tests/test_core.py (hunk 1 above)
Comment (patrys) [reply saleor/graphql/core/tests/test_core.py:257]: The assertion is that calling this method does not raise an exception.

>When a valid JPEG file is uploaded, then `validate_image_file` does not raise.

Comment (IKarbowiak) [reply saleor/graphql/core/tests/test_core.py:257]: So from my perspective, we are checking if `validate_image_file` is not raising an error, so it's the method that we are testing. 🤔

Comment (patrys) [reply saleor/graphql/core/tests/test_core.py:257]: Ok, what is the test then? "When `validate_image_file` is called, then… `pass`?"

The old code checked if a function that never returns returned nothing.

Comment (IKarbowiak) [reply saleor/graphql/core/tests/test_core.py:257]: Ok, I get your point of view. We sometimes use `# when` and `# then` together, but it looks more clear.

**Review by IKarbowiak — APPROVED** [review]

**Comment by patrys** [comment]
@maarcingebala Removed that part of `.editorconfig` as it was never standardized and is not widely supported.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**